x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/rest/action/RestCatTransformAction.java

```diff
@@ -0,0 +1,231 @@
+        }
+
+        GetTransformAction.Request request = new GetTransformAction.Request(id);
+        request.setAllowNoResources(restRequest.paramAsBoolean(ALLOW_NO_MATCH.getPreferredName(), true));
+
+        GetTransformStatsAction.Request statsRequest = new GetTransformStatsAction.Request(id);
+        statsRequest.setAllowNoMatch(restRequest.paramAsBoolean(ALLOW_NO_MATCH.getPreferredName(), true));
+
+        return channel -> client.execute(GetTransformAction.INSTANCE, request, new RestActionListener<>(channel) {
+            @Override
+            public void processResponse(GetTransformAction.Response response) {
+                client.execute(GetTransformStatsAction.INSTANCE, statsRequest, new RestResponseListener<>(channel) {
+                    @Override
+                    public RestResponse buildResponse(GetTransformStatsAction.Response statsResponse) throws Exception {
+                        return RestTable.buildResponse(buildTable(response, statsResponse), channel);
+                    }
+                });
+            }
+        });
+    }
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_cat/transform\n");
+        sb.append("_cat/transform/{" + TransformField.TRANSFORM_ID + "}\n");
+    }
+
+    @Override
+    protected Table getTableWithHeader(RestRequest unused) {
+        return getTableWithHeader();
+    }
+
+    private static Table getTableWithHeader() {
+        return new Table()
+            .startHeaders()
+            // Transform config info
+            .addCell("id", TableColumnAttributeBuilder.builder("the id").build())
+            .addCell("create_time",
+                TableColumnAttributeBuilder.builder("transform creation time")
+                    .setAliases("ct", "createTime")
+                    .build())
+            .addCell("version",
+                TableColumnAttributeBuilder.builder("the version of Elasticsearch when the transform was created")
+                    .setAliases("v")
+                    .build())
+            .addCell("source_index",
+                TableColumnAttributeBuilder.builder("source index")
+                    .setAliases("si", "sourceIndex")
+                    .build())
+            .addCell("dest_index",
+                TableColumnAttributeBuilder.builder("destination index")
+                    .setAliases("di", "destIndex")
+                    .build())
+            .addCell("description",
+                TableColumnAttributeBuilder.builder("description")
+                    .setAliases("d")
+                    .build())
+            .addCell("transform_type",
+                TableColumnAttributeBuilder.builder("batch or continues transform")
+                    .setAliases("sc")
+                    .build())
+            .addCell("frequency",
+                TableColumnAttributeBuilder.builder("frequency of transform")
+                    .setAliases("f")
+                    .build())
```

Comment:
We should also add `PivotConfig#getMaxPageSearchSize()`. This should be accessible in the transform config under `TransformConfig#getPivotConfig`.

It's default value is `500`.

Reply:
Done.